corehq/apps/cleanup/management/commands/deid_case_data.py

```diff
@@ -0,0 +1,135 @@
+    ],
+    'person': [
+        "phone_number",
+        "contact_phone_number",
+        "name",
+        "date_primary_admit",
+        "aadhar_number",
+        "mcts_id",
+        "dob",
+        "raw_aadhar_string",
+        "mcp_id",
+        "rch_id",
+        "date_death",
+        "date_last_private_admit",
+        "bank_account_number",
+        "referral_reached_date",
+        "time_birth",
+    ],
+    'child_health': ["name"],
+    'measurement': ["name"],
+    'tasks': ["name"],
+    'commcare-user': [
+        "phone_number",
+        "helpdesk_phone_number",
+        "dob_aww",
+        "contact_phone_number",
+        "name",
+        "ls_name",
+        "ls_phone_number",
+        "dimagi_username",
+        "email",
+    ],
+    'ccs_record': [
+        "phone_number",
+        "contact_phone_number",
+        "name",
+        "date_primary_admit",
+        "aadhar_number",
+        "mcts_id",
+        "dob",
+        "raw_aadhar_string",
+        "mcp_id",
+        "rch_id",
+        "date_death",
+        "date_last_private_admit",
+        "bank_account_number",
+        "referral_reached_date",
+        "time_birth",
+    ],
+}
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('database')
+
+    def handle(self, domain, database, **options):
+        assert database in get_db_aliases_for_partitioned_query()
+        _create_function(database)
+
+        with connections[database].cursor() as cursor:
+            cursor.execute(
+                "select min(id) as min_id, max(id) as max_id from form_processor_commcarecasesql"
+            )
+            result = fetchone_as_namedtuple(cursor)
+            global_min_id, global_max_id = result.min_id, result.max_id
+
+        print("Data range: %s to %s" % (global_min_id, global_max_id))
+        rows_to_update = 10000
+        batches = [
+            (min, min + rows_to_update)
+            for min in range(global_min_id, global_max_id, rows_to_update)
+        ]
+        for case_type, fields in DEID_FIELDS.items():
+            print("Updating '%s' data" % case_type)
+            keys = '{"%s"}' % '", "'.join(fields)  # '{"f1","f2",...}'
+            values = '{%s}' % ','.join(['""'] * len(fields))  # '{"","",...}'
+            for batch in with_progress_bar(batches, oneline=True):
+                min_id, max_id = batch
+                with connections[database].cursor() as cursor:
+                    cursor.execute("""
+                        UPDATE form_processor_commcarecasesql
+                        SET case_json = json_object_set_keys(case_json::json, %s::TEXT[], %s::TEXT[])
+                        WHERE domain = %s and type = %s AND id > %s AND id <= %s;
+                    """, [keys, values, domain, case_type, min_id, max_id])
+
+                min_id += rows_to_update
+                max_id += rows_to_update
+
+
+def _create_function(db_name):
```

Comment:
any reason you chose to do this here instead of in a migration? 

i guess it's not needed on all envs...

Reply:
This is a once off command that I'm already running. Not sure we should event commit it to the repo since I don't think we'd ever use the approach on live data.

Reply:
+1 we should not commit this